### PR TITLE
Integrate Alice + diagnostic overlay fix

### PR DESCRIPTION
**Pull request by takla21**
GitHub Issue: #

## Proposed Changes

 - [x] Bug fix
 - [x] Feature
 - [ ] Code style update (formatting)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build or CI related changes
 - [ ] Documentation content changes
 - [ ] Other, please describe:

## Description

In this PR, I've integrated Alice in the template to track logs while in-app. I've also took the opportunity to fix the dismiss diagnostic overlay behavior so it's not permanent (just like in the unoapplicationtemplate)

## Impact on version

- [ ] **Major**
  - The template structure was changed.
- [x] **Minor**
  - New functionalities were added.
- [ ] **Patch**
  - A bug in behavior was fixed.
  - Documentation was changed.

### Always applicable
No matter your changes, these checks always apply.
- [x] Your conventional commits are aligned with the **Impact on version** section.
- [x] Updated [CHANGELOG.md](../CHANGELOG.md).
  - Use the latest Major.Minor.X header if you do a **Patch** change.
  - Create a new Major.Minor.X header if you do a **Minor** or **Major** change.
  - If you create a new header, it aligns with the **Impact on version** section and matches what is generated in the build pipeline.
- [x] Documentation files were updated according with the changes.
- [x] Images about the template are referenced from the [wiki](https://github.com/nventive/UnoApplicationTemplate/wiki/Images) and added as images in this git.
- [x] TODO comments are hints for next steps for users of the template and not planned work.

### Contextual
Based on your changes these checks may not apply.
- [ ] Automated tests for the changes have been added/updated.
- [x] Tested on all relevant platforms

### CODE REVIEW

File: src/app/pubspec.yaml [hunk 1]
```diff
@@ -27,6 +27,11 @@ dependencies:
   logger: ^2.2.0
   path_provider: ^2.1.3
   share_plus: ^9.0.0
+  alice: ^0.4.1
+
+dependency_overrides:
```
Comment (takla21) [inline src/app/pubspec.yaml:32]: add a comment that explains why we're doing this

File: src/app/lib/access/logger/alice_output.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import 'package:alice/alice.dart';
+import 'package:flutter/foundation.dart';
+import 'package:logger/logger.dart';
+
+class AliceOutput extends LogOutput {
```
Comment (takla21) [inline src/app/lib/access/logger/alice_output.dart:5]: Mention alice in documentation

File: src/app/lib/access/diagnostics/diagnostics_repository.dart [hunk 3]
```diff
@@ -1,36 +1,48 @@
...
+  Future disableDiagnostics();
 }
 
 /// Implementation of [DiagnosticRepository].
 final class _DiagnosticsRepository implements DiagnosticsRepository {
   /// The key used to store the diagnostic dismissed state in shared preferences.
-  final String _diagnosticDismissedKey = 'diagnosticDismissed';
+  final String _diagnosticDisabledKey = 'diagnosticDisabled';
+
+  /// Whether we want the diagnostics to be accessible by default.
```
Comment (takla21) [inline src/app/lib/access/diagnostics/diagnostics_repository.dart:20]: maybe rectify current diagnostic documentation

File: src/app/lib/business/logger/logger_manager.dart [hunk 4]
```diff
@@ -11,7 +13,8 @@ import 'package:share_plus/share_plus.dart';
 
 /// Service that manages everything related to logging.
 abstract interface class LoggerManager {
-  factory LoggerManager(LoggerRepository loggerRepository) = _LoggerManager;
```
Comment (takla21) [inline src/app/lib/business/logger/logger_manager.dart:14]: test on all platforms

File: src/app/lib/access/logger/alice_output.dart (hunk 2 above)
Comment (takla21) [inline src/app/lib/access/logger/alice_output.dart:5]: add doc

File: doc/Diagnostics.md [hunk 5]
```diff
@@ -13,7 +13,8 @@ This is useful when you want to see something happening live.
 The default buttons include commands such as the following.
 - **Expand/Minimize** opens or closes the expanded view of the overlay.
 - **Move** moves the overlay left or right.
-- **X** hides the overlay.
+- **X** hides the overlay for the remaining of the app cycle.
+    - If you want to permanently hide the diagnostic, go to the environment section within the expanded overlay.
```
Comment (jeanplevesque) [inline doc/Diagnostics.md:17]: You should probably also mention the environment variable

**Review by jeanplevesque — APPROVED** [review]

File: src/app/lib/access/logger/alice_output.dart [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import 'package:alice/alice.dart';
+import 'package:flutter/foundation.dart';
+import 'package:logger/logger.dart';
+
+/// Custom implementation of [LogOutput] to add [Alice] logs writting support.
+class AliceOutput extends LogOutput {
```
Comment (Soap-141) [inline src/app/lib/access/logger/alice_output.dart:6]: final class.

File: src/app/lib/app_router.dart [hunk 7]
```diff
@@ -9,7 +9,7 @@ import 'package:get_it/get_it.dart';
 import 'package:go_router/go_router.dart';
 import 'package:logger/logger.dart';
 
-final _rootNavigatorKey = GlobalKey<NavigatorState>();
+final rootNavigatorKey = GlobalKey<NavigatorState>();
```
Comment (Soap-141) [inline src/app/lib/app_router.dart:12]: make it private may be?

File: src/app/lib/business/logger/logger_manager.dart [hunk 8]
```diff
@@ -62,7 +66,7 @@ final class _LoggerManager implements LoggerManager {
   @override
   bool hasConfigurationBeenChanged = false;
 
-  _LoggerManager(this._loggerRepository);
+  _LoggerManager(this._loggerRepository, this._alice);
```
Comment (Soap-141) [inline src/app/lib/business/logger/logger_manager.dart:69]: Probably should use the other way of setting them, normally we don't expose private member with 'this'.

File: src/app/lib/app_router.dart (hunk 7 above)
Comment (takla21) [reply src/app/lib/app_router.dart:12]: it's needed outside of this file. Especially so alice inspector works

File: src/app/lib/presentation/diagnostic/environment_diagnostic_widget.dart [hunk 9]
```diff
@@ -1,16 +1,47 @@
...
+
+  EnvironmentDiagnosticWidget({super.key});
+
+  void _disableDiagnostics(BuildContext context) async {
+    var hasUserAgreed = await showDialog<bool>(
+      context: context,
+      builder: (ctx) => AlertDialog.adaptive(
+        title: const Text("Disable diagnostic"),
+        content: const Text(
+            "Are you sure you want to disable the diagnostic? If you proceed, you will need to reinstall the app to regain access to it"),
```
Comment (Lee31416) [inline src/app/lib/presentation/diagnostic/environment_diagnostic_widget.dart:20]: Missing comma at the end of the comment

**Review by Lee31416 — APPROVED** [review]

**Review by Soap-141 — APPROVED** [review]
Just see comments, good job!

**Pull request merged**